pkg/bundlereader/resources.go

```diff
@@ -152,9 +152,17 @@ func mergeGenericMap(first, second *fleet.GenericMap) *fleet.GenericMap {
 
 // addRemoteCharts gets the chart url from a helm repo server and returns a `directory` struct.
 // For every chart that is not on disk, create a directory struct that contains the charts URL as path.
-func addRemoteCharts(directories []directory, base string, charts []*fleet.HelmOptions, auth Auth) ([]directory, error) {
+func addRemoteCharts(directories []directory, base string, charts []*fleet.HelmOptions, auth Auth, helmRepoUrlRegex string) ([]directory, error) {
 	for _, chart := range charts {
 		if _, err := os.Stat(filepath.Join(base, chart.Chart)); os.IsNotExist(err) || chart.Repo != "" {
+			shouldAddAuthToRequest, err := shouldAddAuthToRequest(helmRepoUrlRegex, chart.Repo)
+			if err != nil {
+				return nil, err
+			}
+			if !shouldAddAuthToRequest {
```

Comment:
I wonder if it's easier to read if we return a "removeAuth", but it reading it again I think this is fine.